src/wp-includes/class-wp-block-templates-registry.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * Templates registry functions.
+ *
+ * @package WordPress
+ * @since 6.7.0
+ */
+
+/**
+ * Core class used for interacting with templates.
+ *
+ * @since 6.7.0
+ */
+final class WP_Block_Templates_Registry {
+	/**
+	 * Registered templates, as `$name => $instance` pairs.
+	 *
+	 * @since 6.7.0
+	 * @var WP_Block_Template[] $registered_block_templates Registered templates.
+	 */
+	private $registered_templates = array();
+
+	/**
+	 * Container for the main instance of the class.
+	 *
+	 * @since 6.7.0
+	 * @var WP_Block_Templates_Registry|null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Registers a template.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param string $template_name Template name including namespace.
+	 * @param array  $args          Optional. Array of template arguments.
+	 * @return WP_Block_Template|false The registered template on success, or false on failure.
+	 */
+	public function register( $template_name, $args = array() ) {
+
+		$template = null;
+
+		$error_message = '';
+		$error_code    = '';
+
+		if ( ! is_string( $template_name ) ) {
+			$error_message = __( 'Template names must be strings.' );
+			$error_code    = 'template_name_no_string';
+		} elseif ( preg_match( '/[A-Z]+/', $template_name ) ) {
+			$error_message = __( 'Template names must not contain uppercase characters.' );
+			$error_code    = 'template_name_no_uppercase';
+		} elseif ( ! preg_match( '/^[a-z0-9-]+\/\/[a-z0-9-]+$/', $template_name ) ) {
+			$error_message = __( 'Template names must contain a namespace prefix. Example: my-plugin//my-custom-template' );
+			$error_code    = 'template_no_prefix';
+		} elseif ( $this->is_registered( $template_name ) ) {
+			/* translators: %s: Template name. */
+			$error_message = sprintf( __( 'Template "%s" is already registered.' ), $template_name );
+			$error_code    = 'template_already_registered';
+		}
+
+		if ( $error_message ) {
+			_doing_it_wrong(
+				__METHOD__,
+				$error_message,
+				'6.7.0'
+			);
+			return new WP_Error( $error_code, $error_message );
```

Comment:
Return value is expected to be `'WP_Block_Template|false'`, `'WP_Error'` returned.

Reply:
True, that was a leftover from a previous refactor. Thanks for noticing! Fixed in 32bb1e2a5043e1af9754371de0245de7d0ae0827.